north/src/runtime/npk/linux.rs

```diff
@@ -172,7 +172,17 @@ async fn install_internal(
         Manifest::from_str(&manifest)?
     };
     debug!("Manifest loaded for \"{}\"", manifest.name);
-    if let Some(resources) = &manifest.resources {
+
+    let resources: Vec<String> = manifest
+        .mounts
+        .iter()
+        .filter_map(|m| match m {
+            Mount::Resource { name, version, .. } => Some(format!("{} ({})", name, version)),
+            _ => None,
+        })
+        .collect();
+
+    if !resources.is_empty() {
         debug!("Referencing {} resources:", resources.len());
         for res in resources {
```

Comment:
Here's no need to collect with some allocation. The two iterations can be easily merged.

Reply:
The problem here is that if the behavior is not to be changed, it is necessary to know if there are resources in the manifest. Unfortunately there is no way to know without iterating the mounts array. If it is fine to always print the debug string in line 186 independently of the presence of resources, then we can do this in one pass.